### PR TITLE
Quiltcore java preview

### PR DESCRIPTION
**Pull request by drernie**
Ernie's updates for Sep 6

### CODE REVIEW

**Comment by drernie** [comment]
@dimaryaz Are you okay with me merging this into your quiltcore-java-demo branch? Or should I merge it to master, and have you pull from there?

**Comment by dimaryaz** [comment]
Reviewing it now, but you can merge into my branch if you want. For master, I think it's good to review everything and squash all the commits.

File: plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy [hunk 1]
```diff
@@ -136,33 +130,46 @@ class QuiltPackage {
...
-        List<String> result = []
-        final String[] children = subfolder.list().sort()
-        //log.debug("relativeChildren[${base}] $children")
-        for (String pathString : children) {
-            def relative = pathString.replace(base, '')
-            result.add(relative)
+        Manifest manifest = packageManifest()
+        // get names of entries
+        List<String> children = manifest.getEntries().keySet().collect { String key ->
+            if (key.startsWith(subpath)) {
```
Comment (dimaryaz) [inline plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy]: Make sure that this handles "/" properly? I.e., that filtering for "/a" includes "/a/b", but not "/ab"?

File: plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy [hunk 2]
```diff
@@ -233,23 +237,19 @@ class QuiltPackage {
...
-        S3PhysicalKey registryPath = new S3PhysicalKey(bucket, '', null)
-        Registry registry = new Registry(registryPath)
-        Namespace namespace = registry.getNamespace(packageName)
-
+        Namespace namespace = packageNamespace()
         Manifest.Builder builder = Manifest.builder()
-
         Files.walk(packageDest()).filter(f -> Files.isRegularFile(f)).forEach(f -> {
-            System.out.println(f)
+            println(f)
```
Comment (dimaryaz) [inline plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy]: Should just drop it - this was for debugging. (Or make it `log.debug`)

File: plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy (hunk 1 above)
Comment (drernie) [reply plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy]: Can you clarify?  Is a leading "/" required? Do we only require the full folder for absolute subpaths?
Specifically, what's the expected result for these subpaths when the package contains ".ipynb_checkpoints/data_products-checkpoint.ipynb"?

```
        subpath               | expected_size
        ''                    | 8
        '.ipynb_checkpoints'  | 1
        '/.ipynb_checkpoints' | 1
        '.ipynb'              | 1
        '/.ipynb'             | 0
```

Comment (drernie) [reply plugins/nf-quilt/src/main/nextflow/quilt/jep/QuiltPackage.groovy]: I am very confused by this. I rewrote the test to make sense for me, which is:
```
        where:
        subpath               | expected_size
        ''                    | 8
        '.ipynb_checkpoints'  | 1
        '.ipynb_checkpoints/' | 1
        '.ipynb'              | 1
        '.ipynb/'             | 0
```

**Pull request merged**